lambdas/retrieve-cri-credential/src/main/java/uk/gov/di/ipv/core/retrievecricredential/RetrieveCriCredentialHandler.java

```diff
@@ -284,7 +294,10 @@ private Map<String, Object> processVerifiableCredentials(
 
             sendIpvVcReceivedAuditEvent(auditEventUser, vc, isSuccessful);
 
-            submitVcToCiStorage(vc, clientOAuthSessionItem.getGovukSigninJourneyId(), ipAddress);
+            submitVcToCiStorage(vc, govukSigninJourneyId, ipAddress);
+            if (postMitigatingVcs) {
```

Comment:
I think, we can fetch a list and then one call for all vc for performance.
I suggestion, can move the postMitigatingVc codes to a separate method and call it.

Reply:
Definitely one to consider for the future but
- we cater for multiple vc from a CRI but only get a single one in practice as far as I know
- all the other actions are per vc - audit, putContraIndicator, persisting to vc storage, so changing the granularity for postMitigations feels wrong